_common

```diff
@@ -67,9 +69,9 @@ runcurl() {
     response=$(curl -w "\n%{http_code}\n" "$@" 2>&1)
     rc=$?
     set -e
-    [[ "$rc" != 0 ]] && warn "curl ($(caller)): Error fetching ($*): $response" && return 1
+    [[ "$rc" == 0 ]] || die "curl ($(caller)): Error fetching ($*): $response"
     status_code=$(echo "$response" | tail -1)
-    [[ "$status_code" != 200 ]] && warn "curl ($(caller)): Error fetching url ($*): Got Status $status_code" && return 1
+    [[ "$status_code" == 200 ]] || die "curl ($(caller)): Error fetching url ($*): Got Status $status_code"
```

Comment:
but calling die is having a different effect than just returning from this function. I have not checked all calls of runcurl but if you want to replace the usage here then we should make sure it's effectively having the same effect

Reply:
That's a very good point.
`exit` is not same as `return`, though with `set -e` returning non-zero could trigger an exit depending on how it was called.

This inspires me to create a test coverage facility.